### ISSUE TITLE
Compiled wheels

### ISSUE THREAD

**Issue by kislyuk** [ICRAR/crc32c#2]
Hi, thank you for publishing this package, it's very useful and elegant, and the performance is phenomenal.

It would be great if this package could be distributed on PyPI with pre-compiled wheels, so it can be installed without a build toolchain. Would you be open to configuring the release script to publish such wheels? You could use Travis to build Mac OS and Linux wheels, and Appveyor to build Windows wheels.

I may be able to help out if you would be willing to consider a PR for this.

**Comment by rtobar** [comment]
Hi,

Thanks for the feedback on the package, I'm happy to learn it's being useful to more people :)

Regarding the binary wheels, that's something I had though about on the past too. The AppVeyor builds already produce binary wheels and save them as artifacts actually, but I never pushed myself to configure Travis to do something similar, or to upload neither of them automatically to PyPI.

I'm experimenting a bit now, and I think I found a reasonable solution to produce proper `manylinux` binary wheels on Linux based on the `manylinux` container, which itself is based on Centos5 (thus providing the maximum level of compatibility across Linux distributions). I've manually upload the wheels (and Windows') for version 1.4 for the time being, and updated our TravisCI configuration to generate and upload `manylinux` wheels in future releases.

We don't generate many releases for this package, so I wouldn't personally want to invest too much more effort on automating the rest of the process. You are more than free to help out with any further improvements on the MacOS/Windows side if you want though, and I'd definitely consider merging changes if they look good (although it's obviously difficult to prove they are correct until one runs the CI jobs).

**Comment by kislyuk** [comment]
Thanks for merging #3 - do you have any plans for uploading Mac OS wheels? The wheels can be built on the Travis builds but then they need to be either published directly in those builds, or uploaded as artifacts to an S3 bucket that you own, so that you can then publish them.

**Comment by rtobar** [comment]
I've generated and uploaded two wheels for MacOS, but not based on the Travis build (yet). For the time being I generated them manually on a Mac machine I have at hand with MacOS 10.11, using the system python for the 2.7 wheel, and homebrew's python3.6. This should target a wide enough audience for the time being I think, but I will definitely (in the future) address the automatic upload of wheels from the Travis MacOS builds.
